util/reflect_test.go

```diff
@@ -568,6 +582,62 @@ func TestIsValueTypeComaptible(t *testing.T) {
 	}
 }
 
+type derivedByteSlice []byte
+
+func TestInsertIntoStruct(t *testing.T) {
+	type BasicStruct struct {
+		ByteSliceField derivedByteSlice
+	}
+
+	tests := []struct {
+		desc         string
+		parentStruct interface{}
+		fieldName    string
+		fieldValue   interface{}
+		wantVal      interface{}
+		wantErr      string
+	}{
```

Comment:
Can we add a test here that covers the `[]uint8` case that I mentioned above? Thanks for adding this coverage!

Reply:
Added test case for corresponding scenario.

Reply:
Perfect, thanks.